syntax = "proto3";

package processeddata;

message ProcessedData {
    uint32 sequence_num = 1;
    double rx_sample_rate = 2;
    double output_sample_rate = 3;
    repeated OutputDataSet outputdataset = 4; // 1 set for every frequency
    double processing_time = 5;
    double initialization_time = 6;
    double sequence_start_time = 7;
    string rf_samples_location = 8;
    uint32 filter_rolloff_samples = 9;

    message OutputDataSet {
        repeated ComplexData mainacf = 1;
        repeated ComplexData intacf = 2;
        repeated ComplexData xcf = 3;
        uint32 numberofbeams = 4;
        repeated BeamData beamformedsamples = 5;
        repeated DebugData debugsamples = 6;
        uint32 slice_id = 7;


        message BeamData {
            uint32 beamnum = 1;
            repeated ComplexData mainsamples = 2;
            repeated ComplexData intfsamples = 3;
        }

        message DebugData {
            string stagename = 1;
            repeated AntennaData antennadata = 2;
            message AntennaData {
                repeated ComplexData antennasamples = 1;
            }
        }

        message ComplexData {
            float real = 1;
            float imag = 2;
        }
    }
}
